tests/integration/records/table_validator.py

```diff
@@ -155,9 +203,28 @@ def validate_data_values(self, variant, schema_name, table_name):
             assert (ret['timestamp'] == datetime.datetime(2000, 1, 2, 12, 34, 56, 789012)),\
                 f"ret['timestamp'] was {ret['timestamp']}"
 
-        if (self.variant_doesnt_support_timezones(variant) and
+        if (self.loaded_from_dataframe() and
+            self.variant_doesnt_support_timezones(load_variant) and
            not self.database_default_store_timezone_is_us_eastern()):
-            # Example date that we'd be loading:
+            #
+            # In this case, we correctly tell Pandas that we have are
+            # at noon:34 US/Eastern, and tell Pandas to format the
+            # datetime format.  Unfortunately, if you don't specify a
+            # timezone as part of that format, Pandas just prints the
+            # TZ-naive hour.
+            #
+            utc_hour = 12
```

Comment:
This case was was sort of handled by coincidence before, and I think was confusing things quite a bit when debugging test failures.